makefile/Makefile

```diff
@@ -46,39 +46,27 @@ INSITU ?= 0
 # 0: disbaled, 1: enabled
 TBB ?= 0
 
-# clang's address sanitizer 
+# clang's address sanitizer
 ADDRESS_SANITIZER ?= 0
 
-# clang's thread sanitizer 
+# clang's thread sanitizer
 THREAD_SANITIZER ?= 0
 
 # select vectorization mode:
-VECTORIZE_CODE ?= SSE
-# AOS      - LegacyCellProcessor
-# SSE      - VectorizedCellProcessor with SSE intrinsics (default)
-# AVX      - VectorizedCellProcessor with AVX intrinsics
-# AVX2     - VectorizedCellProcessor with AVX2 intrinsics
-# KNL_MASK - VectorizedCellProcessor with KNL intrinsics and masking
-# KNL_G_S  - VectorizedCellProcessor with KNL intrinsics and gather-scatter (not fully optimized yet in this branch)
-# SKX_MASK - VectorizedCellProcessor with SKX intrinsics and masking
-# SKX_G_S  - VectorizedCellProcessor with SKX intrinsics and gather-scatter (not fully optimized yet in this branch)
-### note to the SKX variants: Uses only AVX512F. Only the flags passed to the compiler assume SKX, but not the used intrinsics code.
-# SOA      - VectorizedCellProcessor with no intrinsics
-# (Support dropped!) KNC_MASK - VectorizedCellProcessor with KNC intrinsics and masking
-# (Support dropped!) KNC_G_S  -VectorizedCellProcessor with KNC intrinsics and gather-scatter (not fully optimized yet in this branch)
 # see "make help_vect"
-vec_options := AOS SSE AVX AVX2 KNC_MASK KNC_G_S KNL_MASK KNL_G_S SKX_MASK SKX_G_S SOA SSEAMD AVXAMD
-ifeq (, $(filter $(VECTORIZE_CODE), $(vec_options)))
+VECTORIZE_CODE ?= SSE
+vec_options := NOVEC SSE AVX AVX2 KNL_MASK KNL_G_S SKX_MASK SKX_G_S SSEAMD AVXAMD
+deprecated_vec_options := KNC_MASK KNC_G_S
```

Comment:
remove knc line

Reply:
I kept those because of the "Someone else merge these checks" comment. Do you not want to keep an option to throw special errors for deprecated vecs?

Reply:
ok